files/job_wrapper.sh

```diff
@@ -24,25 +24,51 @@ populate(){
     $_CONDOR_SCRATCH_DIR/parrot_static_run -T 30 cp --no-clobber -r "/chirp/CONDOR/$reana_workflow_dir" "$_CONDOR_SCRATCH_DIR/$parent"
 }
 
-# Discover singularity binary path
-# by trying different methods
-find_singularity(){
-    # Method 1: Find in environment
-    singularity_path="$(which singularity 2>/dev/null)"
-    if [ "x$singularity_path" != "x" ]; then return 0; fi
-    
-    # Method 2: Use module, then find in environment
-    alias module 2>/dev/null
-    if [ $? == 0 ];then
-        MODULE_LIST=(singularity tacc-singularity)
-        for var in ${MODULE_LIST[*]}; do
-            module load $var 2>/dev/null
-            singularity_path="$(which singularity 2>/dev/null)"
-            if [ "x$singularity_path" != "x"]; then return 0; fi
-        done
-    fi
+# Discover the container technology available.
+# Currently searching for: Singularity or Shifter.
+# Returns 0: Successful discovery of a container
+#         1: Couldn't find a container
+find_container(){
+    declare -a search_list=("singularity" "shifter")
+    declare -a module_list=(singularity tacc-singularity)
+    declare -A found_list=()
```

Comment:
Note the use of associative arrays means we will need bash >= 4.x. I can't see why that would be a problem on any Linux-based OS though, considering the release date was about 10 years ago (Mac is the only OS I know that still uses bash 3.x), but we should probably put a comment at the top of the script, just in case.

Reply:
`Blue Waters` nodes are `GNU bash, version 3.2.51(1)-release (x86_64-suse-linux-gnu)` so there is an issue. Will consider other options and rework that section of code. Thanks for bringing this up.

Reply:
Resolved in 3951c0478931f9333c349efacbab5daed2833c1f